build_latest.sh

```diff
@@ -37,6 +37,88 @@ set_version "$1"
 vm="$2"
 package="$3"
 
+# Get the image build time stored in the respective build_time array passed as arg
+function get_image_build_time() {
+	if ! declare -p "$1" &>/dev/null; then
+		return;
+	fi
+
+	# shellcheck disable=SC2154,SC1083
+	btime=$(btarray=$1[${current_arch}]; eval btarch=\${"$btarray"}; echo "${btarch}");
+
+	echo "${btime}"
+}
+
+# Check if we need to do a docker build
+# Build is needed only if one of the following criteria is met
+# 1. If no such docker image exists currently
+# 2. If the base OS docker image was recently re-built
+# 3. If a new Adopt build is found
+# 4. On any other error condition
+function check_build_needed() {
```

Comment:
Could each of these cases be put into their own function?

Reply:
Converted to functions now. Thanks !